custom_components/localtuya/core/ha_entities/__init__.py

```diff
@@ -227,9 +227,16 @@ def get_dp_values(dp: str, dps_data: dict, req_info: CLOUD_VALUE = None) -> dict
 
     # ENUM Values: range: list of values.
     if dp_values and dp_type == DPType.ENUM:
-        dp_values["min"] = dp_values.get("range", [])[0]  # first value
-        dp_values["max"] = dp_values.get("range", [])[-1]  # Last value
-        dp_values["range"] = convert_list(dp_values.get("range"), req_info)
+        range_values = dp_values.get("range", []) 
+        if range_values:  # Check if range is not null or empty
+            dp_values["min"] = range_values[0]  # First value
+            dp_values["max"] = range_values[-1]  # Last value
+            dp_values["range"] = convert_list(range_values, req_info)
+        else:
+            dp_values["min"] = 0 # Or some default value
+            dp_values["max"] = 0 # Or some default value
+            dp_values["range"] = []  # Or some default value
+            
```

Comment:
```suggestion
        range_values = dp_values.get("range", [])

        dp_values["min"] = range_values[0] if range_values else 0  # first value
        dp_values["max"] = range_values[-1] if range_values else 0  # Last value
        dp_values["range"] = convert_list(range_values, req_info)
```

Convert list will also need to return empty list if range is empty instead of empty string.

https://github.com/xZetsubou/hass-localtuya/blob/4aefbdffa52dacf9ab087c8371e25260a5e2616c/custom_components/localtuya/core/ha_entities/__init__.py#L258-L261

```py
def convert_list(_list: list, req_info: CLOUD_VALUE = str):
    """Return list to dict values."""
    if not _list:
        return []
```